project/frontend/src/components/RestaurantCard.js

```diff
@@ -17,49 +17,110 @@ import Clock from '../assets/clock.svg';
  *     information.
  * @param {!Object<string, *>} props.details An object containing detailed restaurant
  *     information.
+ * @param {?Object<string, *>} props.style An optional style object to be applied
+ *     to the parent container of this component.
+ * @param {boolean=} props.collapsed An optional control over whether this card is in
+ *     a collapsed or full view (default: false).
  */
 function RestaurantCard(props) {
-  const { restaurant, details } = props;
+  const { restaurant, details, style, collapsed = false } = props;
 
-  const photoUrls = details.result.photos.map(
-    (photo) =>
-      'https://maps.googleapis.com/maps/api/place/photo?photoreference=' +
-      photo.photo_reference +
-      '&key=' +
-      process.env.REACT_APP_GOOGLE_API_KEY +
-      '&maxwidth=344'
-  );
+  const photoUrls = details
+    ? details.result.photos.map(
+        (photo) =>
+          'https://maps.googleapis.com/maps/api/place/photo?photoreference=' +
+          photo.photo_reference +
+          '&key=' +
+          process.env.REACT_APP_GOOGLE_API_KEY +
+          '&maxwidth=344'
```

Comment:
nit: you can use a template literal here ``https://maps.googleapis.com/maps/api/place/photo?photoreference=${photo.photo_reference}&key=${process.env.REACT_APP_GOOGLE_API_KEY}&maxwidth=344``
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals